### ISSUE TITLE
W1b: AugmentationSequential silently passes mask, box and keypoint keys through mix children

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4493]
`AugmentationSequential` mixes the image through a mix child and hands every other data key back unchanged, so masks, boxes and keypoints silently desynchronize from the mixed image. The same keys raise `NotImplementedError` when the mix class is called directly, and `RandomMosaic` transforms boxes directly, so the container both hides an error and drops a working handler.

## Observed (`main` `a7e2db6b`, torch 2.14.0, CPU)

| child | key | direct call | inside `AugmentationSequential` |
|---|---|---|---|
| `RandomMixUpV2`, `RandomCutMixV2`, `RandomJigsaw`, `PatchMix` | `mask`, `bbox_xyxy`, `keypoints` | `NotImplementedError` | returned unchanged, image mixed |
| `RandomMosaic` | `bbox_xyxy` | boxes transformed | returned unchanged, image mixed |
| `RandomMosaic` | `mask`, `keypoints` | `NotImplementedError` | returned unchanged |
| `RandomMixUpV2`, `RandomCutMixV2` | `class` | labels returned | `NotImplementedError` from the container |

Only `RandomTransplantation` has a mask branch in the container.

## Mechanism

`MaskSequentialOps.transform`, `BoxSequentialOps.transform` and `KeypointSequentialOps.transform` in `kornia/augmentation/container/ops.py` dispatch on `GeometricAugmentationBase2D`, `GeometricAugmentationBase3D`, `_AugmentationBase`, the containers and `OperationBase`, with a `RandomTransplantation` special case for masks. A `MixAugmentationBaseV2` child matches none of them and reaches the trailing `return input`. `ClassSequentialOps.transform` is the only handler that checks for the mix base, and it raises.

## Repro

```python
import torch, kornia.augmentation as K

torch.manual_seed(0)
x = torch.rand(2, 3, 16, 16)
boxes = torch.tensor([[[0.0, 0.0, 2.0, 2.0]], [[1.0, 1.0, 3.0, 3.0]]])
mask = torch.arange(2.0).reshape(2, 1, 1, 1).expand(2, 1, 16, 16).clone()

out = K.AugmentationSequential(K.RandomMosaic(p=1.0), data_keys=["input", "bbox_xyxy", "mask"])(x, boxes, mask)
print(torch.equal(out[0], x), torch.equal(out[1], boxes), torch.equal(out[2], mask))  # False True True

print(torch.equal(K.RandomMosaic(p=1.0, data_keys=["input", "bbox_xyxy"])(x, boxes)[1], boxes))  # False
K.RandomMixUpV2(p=1.0)(x, mask, data_keys=["input", "mask"])  # NotImplementedError
```

## Expected

Either the container raises for a key the mix child cannot transform, as the direct call does, or it dispatches to the child's own handler where one exists (`RandomMosaic` boxes). Silent passthrough of an annotation next to a mixed image is the same failure shape as #4420 for non-rigid children.

## Docs

The `AugmentationSequential` docstring note said mix augmentations "can only be working with input/image data key"; #4452 rewrites it to state the passthrough and link here. Batch 6d (#4407) covers the mix classes' own data-key contracts.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
